commit 95103c3a332da5be4b100f67bfd946f0ac48989a
Author: Mateusz Jablonski <mateusz.jablonski@intel.com>
Date:   Mon Apr 4 15:23:32 2022 +0000

    Correct getters in generated commands
    
    cast to uint64_t before shifting value
    
    Related-To: NEO-6475
    Signed-off-by: Mateusz Jablonski <mateusz.jablonski@intel.com>

diff --git a/shared/source/generated/gen12lp/hw_cmds_generated_gen12lp.inl b/shared/source/generated/gen12lp/hw_cmds_generated_gen12lp.inl
index 0d1ad5807..ae257bc29 100644
--- a/shared/source/generated/gen12lp/hw_cmds_generated_gen12lp.inl
+++ b/shared/source/generated/gen12lp/hw_cmds_generated_gen12lp.inl
@@ -5262,9 +5262,9 @@ typedef struct tagL3_FLUSH_ADDRESS_RANGE {
 
     inline uint64_t getAddressLow(bool isA0Stepping) const {
         if (isA0Stepping) {
-            return (TheStructure.CommonA0Stepping.AddressLow << ADDRESSLOW_BIT_SHIFT) & uint64_t(0xffffffff);
+            return (static_cast<uint64_t>(TheStructure.CommonA0Stepping.AddressLow) << ADDRESSLOW_BIT_SHIFT) & uint64_t(0xffffffff);
         } else {
-            return (TheStructure.Common.AddressLow << ADDRESSLOW_BIT_SHIFT);
+            return (static_cast<uint64_t>(TheStructure.Common.AddressLow) << ADDRESSLOW_BIT_SHIFT);
         }
     }
 
diff --git a/shared/source/generated/xe_hp_core/hw_cmds_generated_xe_hp_core.inl b/shared/source/generated/xe_hp_core/hw_cmds_generated_xe_hp_core.inl
index d290b733c..b2aea223a 100644
--- a/shared/source/generated/xe_hp_core/hw_cmds_generated_xe_hp_core.inl
+++ b/shared/source/generated/xe_hp_core/hw_cmds_generated_xe_hp_core.inl
@@ -6804,7 +6804,7 @@ typedef struct tagL3_FLUSH_ADDRESS_RANGE {
     }
 
     inline uint64_t getAddressLow() const {
-        return (TheStructure.Common.AddressLow << ADDRESSLOW_BIT_SHIFT);
+        return (static_cast<uint64_t>(TheStructure.Common.AddressLow) << ADDRESSLOW_BIT_SHIFT);
     }
 
     inline void setAddressHigh(const uint64_t value) {
@@ -6863,7 +6863,7 @@ struct L3_CONTROL_POST_SYNC_DATA {
         TheStructure.Common.Address = value >> ADDRESS_BIT_SHIFT;
     }
     inline uint64_t getAddress() const {
-        return TheStructure.Common.Address << ADDRESS_BIT_SHIFT;
+        return static_cast<uint64_t>(TheStructure.Common.Address) << ADDRESS_BIT_SHIFT;
     }
     inline void setImmediateData(const uint64_t value) {
         TheStructure.Common.ImmediateData = value;
